test/src/test/java/org/corfudb/utils/LockStoreTest.java

```diff
@@ -0,0 +1,239 @@
+        Table<LockDataTypes.LockId, LockDataTypes.LockData, Message> lockTable =
+            corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, LOCK_TABLE_NAME);
+        lockTable.clearAll();
+    }
+
+    // Two runtimes try to renew an acquired lock concurrently.  Verify that
+    // only the current owner of the lock can renew it.
+    @Test
+    public void testRenewalBySameNode() throws Exception {
+        LockDataTypes.LockId lockId = LockDataTypes.LockId.newBuilder()
+            .setLockGroup(LOCK_GROUP).setLockName(LOCK_NAME).build();
+        CorfuRuntime runtime1 = getDefaultRuntime().connect();
+        CorfuRuntime runtime2 = getNewRuntime(getDefaultNode()).connect();
+        CorfuStore corfuStore1 = new CorfuStore(runtime1);
+        CorfuStore corfuStore2 = new CorfuStore(runtime2);
+        LockStore lockStore1 = new LockStore(runtime1, UUID.randomUUID());
+        LockStore lockStore2 = new LockStore(runtime2, UUID.randomUUID());
+
+        for (int i = 0; i < NUM_ITERATIONS; i++) {
+            scheduleConcurrently(f -> {
+                try {
+                    acquired1 = lockStore1.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired1 = false;
+                }
+            });
+
+            scheduleConcurrently(f -> {
+                try {
+                    acquired2 = lockStore2.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired2 = false;
+                }
+            });
+            executeScheduled(2, TIMEOUT, TimeUnit.SECONDS);
+            Assert.assertTrue(acquired1 || acquired2);
+            Assert.assertFalse(acquired1 && acquired2);
+
+            TxnContext txn = corfuStore1.txn(CORFU_SYSTEM_NAMESPACE);
+            LockDataTypes.LockData dataBeforeRenewal =
+                lockStore1.get(lockId, txn).get();
+            txn.commit();
+
+            // Renew the lock
+            scheduleConcurrently(f -> {
+                try {
+                    acquired1 = lockStore1.renew(lockId);
+                } catch (LockStoreException le) {
+                    acquired1 = false;
+                }
+            });
+            scheduleConcurrently(f -> {
+                try {
+                    acquired2 = lockStore2.renew(lockId);
+                } catch (LockStoreException le) {
+                    acquired2 = false;
+                }
+            });
+            executeScheduled(2, TIMEOUT, TimeUnit.SECONDS);
+            Assert.assertTrue(acquired1 || acquired2);
+            Assert.assertFalse(acquired1 && acquired2);
+
+            txn = corfuStore2.txn(CORFU_SYSTEM_NAMESPACE);
+            LockDataTypes.LockData dataAfterRenewal =
+                lockStore1.get(lockId, txn).get();
+            txn.commit();
+
+            Assert.assertTrue(Objects.equals(
+                dataBeforeRenewal.getLeaseOwnerId(),
+                dataAfterRenewal.getLeaseOwnerId()));
+
```

Comment:
We should also verify that the renewal number is bumped

Reply:
Done.  Thanks.

Reply:
Done, thanks.